### PR TITLE
feat(anvil): use Alloy otterscan types

### LINKED ISSUE #8298: replace otterscan types with alloy's otterscan types
### Component

Anvil

### Describe the feature you would like

all of the otterscan types: 
https://github.com/foundry-rs/foundry/blob/e3021017b643e43e42a4c420b636a2ae947bf89f/crates/anvil/src/eth/otterscan/types.rs#L84-L84

should be available in alloy and can be replaced

https://github.com/alloy-rs/alloy/blob/440ee680f9b16c8813507bae501a4b72a3610f05/crates/rpc-types-trace/src/otterscan.rs#L25-L25

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fedemagnani**
## Motivation
Closes #8298 

## Solution
Deprecates `anvil/src/eth/otterscan/types.rs` in favor of Alloy's otterscan types. As recommended by @DaniPopes, logic is moved into `otterscan/api.rs` expanding `Backend` implementations and adding free functions.

### CODE REVIEW

**Comment by DaniPopes** [comment]
Please format using nightly `cargo +nightly fmt`

**Review by mattsse — COMMENTED** [review]
unable to review this with all the `;` changes ;)

**Comment by fedemagnani** [comment]
> unable to review this with all the `;` changes ;)

First commit contains all the relevant changes. 

In the second commit I synced the fork and this caused the fmt check to fail. 
So I pushed the third commit (containing all the `;` fixes) after running `cargo +nightly fmt` which edited modules not involved in my PR. 

Shall I revert the last commit?

**Comment by fedemagnani** [comment]
@mattsse Hopefully better, sorry! :)

File: crates/anvil/src/eth/otterscan/api.rs [hunk 1]
```diff
@@ -1,19 +1,201 @@
...
+                    value: call.value,
+                    input: call.input.0.into(),
+                })
+            }
+            Action::Create(_) | Action::Selfdestruct(_) | Action::Reward(_) => None,
+        })
+        .collect()
+}
+
+impl Backend {
```
[the remark this answers is not in the document]
Comment (fedemagnani) [reply crates/anvil/src/eth/otterscan/api.rs:68]: Sounds good, adding them to EthApi impl

File: crates/anvil/tests/it/otterscan.rs [hunk 2]
```diff
@@ -243,50 +239,45 @@ async fn test_call_ots_trace_transaction() {
 
     let res = api.ots_trace_transaction(receipt.transaction_hash).await.unwrap();
     let expected = vec![
-        OtsTrace {
-            r#type: OtsTraceType::Call,
+        TraceEntry {
+            r#type: format!("{:?}", CallType::Call),
```
[the remark this answers is not in the document]
Comment (fedemagnani) [reply crates/anvil/tests/it/otterscan.rs:243]: I used it bacause I noticed that `alloy_rpc_types::trace::parity::CallType` doesn't implement Display.

File: crates/anvil/src/eth/otterscan/api.rs [hunk 3]
```diff
@@ -1,19 +1,66 @@
...
+}
+
+/// Converts the list of traces for a transaction into the expected Otterscan format, as
+/// specified in the [`ots_traceTransaction`](https://github.com/otterscan/otterscan/blob/develop/docs/custom-jsonrpc.md#ots_tracetransaction) spec
+pub fn batch_build_ots_traces(traces: Vec<LocalizedTransactionTrace>) -> Vec<TraceEntry> {
+    traces
+        .into_iter()
+        .filter_map(|trace| match trace.trace.action {
+            Action::Call(call) => {
+                let ots_type = serde_json::to_string(&call.call_type).unwrap();
```
Comment (DaniPopes) [inline crates/anvil/src/eth/otterscan/api.rs:49]: this is wrong, it writes `\"CALL\"` when it should be `"CALL"`

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/anvil/src/eth/otterscan/api.rs (hunk 3 above)
Comment (fedemagnani) [reply crates/anvil/src/eth/otterscan/api.rs]: @DaniPopes My bad, I didn't check that the type was serializing in lowercase and I forgot about the double quotes. 
However, `CallType` doesn't implement Display, so I'm converting it into `String` in the following way

File: crates/anvil/src/eth/otterscan/api.rs [hunk 4]
```diff
@@ -1,19 +1,75 @@
...
+}
+
+/// Converts the list of traces for a transaction into the expected Otterscan format, as
+/// specified in the [`ots_traceTransaction`](https://github.com/otterscan/otterscan/blob/develop/docs/custom-jsonrpc.md#ots_tracetransaction) spec
+pub fn batch_build_ots_traces(traces: Vec<LocalizedTransactionTrace>) -> Vec<TraceEntry> {
+    traces
+        .into_iter()
+        .filter_map(|trace| match trace.trace.action {
+            Action::Call(call) => {
+                // let ots_type = serde_json::to_string(&call.call_type).unwrap();
```
Comment (DaniPopes) [inline crates/anvil/src/eth/otterscan/api.rs:49]: pls remove comment

**Review by DaniPopes — APPROVED** [review]
lgtm last nit

**Review by DaniPopes — APPROVED** [review]
pending @mattsse

**Comment by DaniPopes** [comment]
@mattsse gm

**Comment by fedemagnani** [comment]
@DaniPopes Do I need to rebase from `master`? I'm afraid it could mess up with the review

**Review by mattsse — APPROVED** [review]
gm!

**Pull request merged**